### PR TITLE
MarkdownBear: Add setting to enforce maximum line length

### PR DESCRIPTION
**Pull request by namanyadav12**
New parameter to enforce maximum allowed line length.

Closes https://github.com/coala/coala-bears/issues/1181

### CODE REVIEW

**Comment by Nosferatul** [comment]
Change shortlog message from commit message to see if it fails on ci

**Comment by sils** [comment]
hm, tests are failing, can you get it to pass?

File: bears/markdown/MarkdownBear.py [hunk 1]
```diff
@@ -39,7 +44,8 @@ def create_arguments(filename, file, config_file,
                          markdown_list_increment: bool=True,
                          markdown_horizontal_rule: str='*',
                          markdown_horizontal_rule_spaces: bool=False,
-                         markdown_horizontal_rule_repeat: int=3):
+                         markdown_horizontal_rule_repeat: int=3,
+                         markdown_maximum_line_length: int=80):
```
Comment (jayvdb) [inline bears/markdown/MarkdownBear.py]: this should be `max_line_length`, which is a commonly used setting by many bears.

**Comment by sils** [comment]
yes we should also deprecate the other settings and remove the markdown_

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala.io/ - http://viperdev.io/ - http://gitmate.com/

**Comment by yukiisbored** [comment]
Looks good to me, but I don't know if the others want to have legacy support.

**Comment by yukiisbored** [comment]
ack https://github.com/coala/coala-bears/pull/1204/commits/7e0c46875a87a0edda3b2cf03c3586c078f311d9

**Comment by Mixih** [comment]
I'm worry about our possible users who may be using the old settings. I think he should be deprecating the settings before flat out breaking them like this. IMO, we still might have users using the old settings... Look into the deprecation decorator and see if it will for for you.

**Comment by Mixih** [comment]
Also, we should have a policy written out for stuff like this, otherwise we could end up having random usability issues when we revise settings.

**Comment by Mixih** [comment]
unack 7e0c468

**Comment by namanyadav12** [comment]
ok @Mixih , i will try the  deprecation decorator.

**Comment by yukiisbored** [comment]
ack https://github.com/coala/coala-bears/pull/1204/commits/977e629f6d16d1d7c5b6c970b6cc565ad1f9b14f

**Comment by sils** [comment]
unack 977e629 (thanks anyway for the review yuki!) What you do here is too complicated for one commit, please split it up: we need one commit for the renaming of all the params and a second commit that changes the bear. Both together are too complicated for us to ensure no bugs sneak in.

**Comment by sils** [comment]
but generally love it :)

**Comment by namanyadav12** [comment]
@sils done.

File: bears/markdown/MarkdownBear.py [hunk 2]
```diff
@@ -39,7 +45,8 @@ def create_arguments(filename, file, config_file,
                          markdown_list_increment: bool=True,
                          markdown_horizontal_rule: str='*',
                          markdown_horizontal_rule_spaces: bool=False,
-                         markdown_horizontal_rule_repeat: int=3):
+                         markdown_horizontal_rule_repeat: int=3,
+                         max_line_length: int=80):
```
Comment (sils) [inline bears/markdown/MarkdownBear.py]: can we have a testcase for this?

File: bears/markdown/MarkdownBear.py [hunk 3]
```diff
@@ -30,87 +31,103 @@ class MarkdownBear:
...
+            'fences': fences,                           # Bool
+            'listItemIndent': list_indent,              # int or "tab"
+                                                        # or "mixed"
+            'looseTable': loose_tables,                 # Bool
+            'spacedTable': spaced_tables,               # Bool
+            'incrementListMarker': list_increment,      # Bool
+            'rule': horizontal_rule,                    # - or * or _
+            'ruleSpaces': horizontal_rule_spaces,       # Bool
+            'ruleRepetition': horizontal_rule_repeat,   # int
+            }
```
Comment (sils) [inline bears/markdown/MarkdownBear.py]: why the indentation change?

**Comment by sils** [comment]
almost done, testcase missing :)

**Comment by namanyadav12** [comment]
@sils added a testcase

**Comment by sils** [comment]
ack https://github.com/coala/coala-bears/pull/1204/commits/57dbcc422bacece92d80310793147601b2deacd1 https://github.com/coala/coala-bears/pull/1204/commits/75d3d4b56a948a178f8c822cbb1ddca54fab5a46

**Comment by sils** [comment]
@rultor merge

**Comment by sils** [comment]
Hey, this is your first contribution to one of the more quality controlled repos, right? Congrats and welcome to the coalaians! :) You've done your first steps to become a coala developer.

Rultor will now automatically release your changes as a prerelease so users can experience the goodness of your changes right now.

We would really appreciate it if you could help us with

- [telling us about your experience](http://coala.io/newform)
- [some code reviews](http://coala.io/reviewing) for other newcomers
- fixing at least [a difficulty/low issue](http://coala.io/low)
- fill our [usability survey](http://coala.io/usability)

If you've done those and tell us about it so we can promote you to a coala developer!

We're all volunteers and we're currently struggling to keep this up. Helping all those newcomers is a lot of work and we do need your help so we can continue this for others!

**Comment by jayvdb** [comment]
Note that we are assuming that this broke MarkdownBear.  See https://github.com/coala/coala-bears/issues/1235

**Pull request merged**